apps/admin-web/src/screens/Events/DataCards/NumberOfEventCard.tsx

```diff
@@ -0,0 +1,229 @@
+  font-weight: bold;
+`;
+const EventNumbersText: AnyStyledComponent = styled.div`
+  font-size: 15px;
+  padding: 0;
+  display: flex;
+  margin-top: -15px;
+`;
+const EventDifference: AnyStyledComponent = styled.div`
+  display: flex;
+  flex: 1;
+  font-size: 30px;
+  font-weight: bold;
+  flex-direction: column;
+`;
+const EventDifferenceText: AnyStyledComponent = styled.div`
+  font-size: 15px;
+
+  display: flex;
+  margin-top: -15px;
+`;
+const EventDifferenceIcon: AnyStyledComponent = styled.div`
+  display: inline-block;
+  padding-left: 20px;
+`;
+const EventDifferenceNumber: AnyStyledComponent = styled.div`
+  display: inline-block;
+  padding-left: 20px;
+`;
+
+const todaysDate = new Date();
+const todaysISO = todaysDate.toISOString();
+
+const NumberOfEventCard: React.SFC<{}> = (): JSX.Element => {
+  const { loading, error, data }: any = useEventsQuery();
+  let events: any;
+  if (loading) {
+    events = errorData;
+  } else if (error) {
+    events = errorData;
+  } else {
+    events = data.events;
+  }
+
+  const [title, setTitle] = useState("This week ");
+  let [eventNumber, difference] = ThisWeekData();
+  let carrot, fromLast;
+
+  const greenRed = {
+    color: "#59595a",
+  };
+
+  const menu = (
+    <Menu>
+      <Menu.Item>
+        <a onClick={() => setTitle("This week ")}>This week</a>
+      </Menu.Item>
+      <Menu.Item>
+        <a onClick={() => setTitle("This month ")}>This month</a>
+      </Menu.Item>
+      {/* <Menu.Item>
+        <a onClick={() => setTitle("6 months ")}>6 months</a>
+      </Menu.Item> */}
+    </Menu>
+  );
+
+  function ThisWeekData() {
+    const thisWeekFilter = events.filter(
+      (a: any) =>
+        new Date(a.dateHosted.split("T")[0]).getTime() -
+          new Date(a.dateHosted.split("T")[0]).getDay() * 86400000 ===
+        new Date(todaysISO.split("T")[0]).getTime() -
+          new Date(todaysISO.split("T")[0]).getDay() * 86400000
+    );
+
+    const lastWeekFilter = events.filter(
+      (a: any) =>
+        new Date(a.dateHosted.split("T")[0]).getTime() -
+          new Date(a.dateHosted.split("T")[0]).getDay() * 86400000 ===
+        new Date(todaysISO.split("T")[0]).getTime() -
+          new Date(todaysISO.split("T")[0]).getDay() * 86400000 -
+          7 * 86400000
+    );
+
+    const amountOfThisWeekEvents = thisWeekFilter.length;
+    const thisWeekDifference = amountOfThisWeekEvents - lastWeekFilter.length;
+
+    return [amountOfThisWeekEvents, thisWeekDifference];
+  }
+
+  function thisMonthData() {
+    const thisMonthFilter = events.filter(
+      (c: any) =>
+        new Date(c.dateHosted.split("T")[0]).getTime() <
+        new Date(todaysISO.split("T")[0]).getTime() + 30 * 86400000
+    );
+    const thisMonthPreviousFilter = thisMonthFilter.filter(
+      (c: any) =>
+        new Date(c.dateHosted.split("T")[0]).getTime() -
+          new Date(todaysISO.split("T")[0]).getTime() >
+        0
+    );
+    const lastMonthFilter = events.filter(
```

Comment:
Can you create a variable for "time frame" and set it to be a number of days? then just make one filter function?